debian/patches/backport-ubuntu-pro-rename.patch

```diff
@@ -0,0 +1,2522 @@
+-
+-import sys
+-from textwrap import dedent
+-
+-from cloudinit import log as logging
+-from cloudinit.config.schema import (
+-    get_schema_doc, validate_cloudconfig_schema)
+-from cloudinit.settings import PER_INSTANCE
+-from cloudinit.subp import prepend_base_command
+-from cloudinit import subp
+-from cloudinit import util
+-
+-
+-distros = ['ubuntu']
+-frequency = PER_INSTANCE
+-
+-LOG = logging.getLogger(__name__)
+-
+-schema = {
+-    'id': 'cc_ubuntu_advantage',
+-    'name': 'Ubuntu Advantage',
+-    'title': 'Install, configure and manage ubuntu-advantage offerings',
+-    'description': dedent("""\
+-        This module provides configuration options to setup ubuntu-advantage
+-        subscriptions.
+-
+-        .. note::
+-            Both ``commands`` value can be either a dictionary or a list. If
+-            the configuration provided is a dictionary, the keys are only used
+-            to order the execution of the commands and the dictionary is
+-            merged with any vendor-data ubuntu-advantage configuration
+-            provided. If a ``commands`` is provided as a list, any vendor-data
+-            ubuntu-advantage ``commands`` are ignored.
+-
+-        Ubuntu-advantage ``commands`` is a dictionary or list of
+-        ubuntu-advantage commands to run on the deployed machine.
+-        These commands can be used to enable or disable subscriptions to
+-        various ubuntu-advantage products. See 'man ubuntu-advantage' for more
+-        information on supported subcommands.
+-
+-        .. note::
+-           Each command item can be a string or list. If the item is a list,
+-           'ubuntu-advantage' can be omitted and it will automatically be
+-           inserted as part of the command.
+-        """),
+-    'distros': distros,
+-    'examples': [dedent("""\
+-        # Enable Extended Security Maintenance using your service auth token
+-        ubuntu-advantage:
+-            commands:
+-              00: ubuntu-advantage enable-esm <token>
+-    """), dedent("""\
+-        # Enable livepatch by providing your livepatch token
+-        ubuntu-advantage:
+-            commands:
+-                00: ubuntu-advantage enable-livepatch <livepatch-token>
+-
+-    """), dedent("""\
+-        # Convenience: the ubuntu-advantage command can be omitted when
+-        # specifying commands as a list and 'ubuntu-advantage' will
+-        # automatically be prepended.
+-        # The following commands are equivalent
+-        ubuntu-advantage:
+-            commands:
+-                00: ['enable-livepatch', 'my-token']
+-                01: ['ubuntu-advantage', 'enable-livepatch', 'my-token']
+-                02: ubuntu-advantage enable-livepatch my-token
+-                03: 'ubuntu-advantage enable-livepatch my-token'
+-    """)],
+-    'frequency': PER_INSTANCE,
+-    'type': 'object',
+-    'properties': {
+-        'ubuntu-advantage': {
+-            'type': 'object',
+-            'properties': {
```

Comment:
Good thinking, I had mistakenly looked over the xenial branch before applying quilt patches and didn't recall that we carried the d/patches/ubuntu-advantage-revert-tip.patch.  I have since dropped that patch from our series and just applied the direct d/patches/backport-ubuntu-pro-config.patch which handles file renames and module renames and cc_ubunutu_pro.schema additions. I"ll update the test verification  in the PR to also exercise the ubuntu-advantage:commands testing.